### ISSUE TITLE
Iris Dataset Analysis Using Decision Tree

### ISSUE THREAD

**Issue by thamizhthamizh725-lgtm** [thamizhthamizh725-lgtm/iris#1]
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score
import matplotlib.pyplot as plt

# 1. Load the Iris dataset
df = pd.read_csv('iris.csv')
print("last 5 rows of dataset:")
print(df.tail())

# 2. Encode the species column (Label Encoding)
df['species'] = df['species'].astype('category').cat.codes

# 3. Split the dataset into features and labels
X = df.drop('species', axis=1)
y = df['species']

# 4. Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

# 5. Train the model
model = DecisionTreeClassifier()
model.fit(X_train, y_train)

# 6. Make predictions
y_pred = model.predict(X_test)

# 7. Evaluate accuracy
accuracy = accuracy_score(y_test, y_pred)
print("\nModel Accuracy:", accuracy)

# 8. Visualize Sepal Length vs Width
plt.scatter(df['sepal_length'], df['sepal_width'])
plt.xlabel('Sepal Length')
plt.ylabel('Sepal Width')
plt.title('Iris Sepal Length vs Sepal Width')
plt.show()
